### DISCUSSION TITLE
Codespaces inactivity

### DISCUSSION THREAD

**Discussion by xoiga123** [community/community#53722] [Codespaces] [answered]
### Select Topic Area

Bug

### Body

Context:
- My organization default timeout for inactivity is 4 hours.
- My personal default timeout for inactivity is 30 minutes.

The problem:
- In my python script, I made sure to print to console every minute so I'm not inactive, but I still get flagged for inactivity and the instance is automatically stopped at around the 4 hours mark, with no message whatsoever except `Stopping instance...`. 
- I even went further and created a cronjob to append characters to a temporary file every minute, but the instance still got stopped (verified the job ran fine for about 4 hours).

**Comment by cwndrws** [comment, marked as answer]
Hi 👋 thanks for writing in! Codespaces is designed to be a cloud-based interactive development envionment. The key term in this case being interactive. It's not really designed to be used as a batch processing tool like you seem to be wanting. GitHub Actions might be a better fit for you use-case since it's designed to run for longer and not be interactive.

As you can see in [the Actions docs](https://docs.github.com/en/actions/learn-github-actions/usage-limits-billing-and-administration) an individual job can run for up to 6 hours and workflow can run for days.

**Reply by xoiga123** [reply]
Gotcha! Thanks for the explanation. Would like to have this on the docs somewhere tho, as you provide beefy instances and I just thought maybe... :rofl:

**Comment by Aaevelion** [comment]
Hi,

I have a big dataset which I want to analyse (a few GBs). I was using codespaces to do this, but unfortunately it is shutting down after some time (the script takes longer than that to complete running successfully--it has been tested before). How should I go about running Actions for this?
